File: core/src/net/natruid/jungle/events/UnitSkillEvent.java
package net.natruid.jungle.events;

import net.natruid.jungle.utils.PathNode;

import java.util.Deque;

public class UnitSkillEvent extends PooledEvent {
    public int unit = -1;
    public int skill = -1;
    public int target = -1;
    public Deque<PathNode> path = null;

    @Override
    public void reset() {
        unit = -1;
        skill = -1;
        target = -1;
        path = null;
    }
}

File: core/src/net/natruid/jungle/events/UnitSkillEvent.kt
package net.natruid.jungle.events

import net.natruid.jungle.utils.Path

class UnitSkillEvent : PooledEvent() {
    var unit = -1
    var skill = -1
    var target = -1
    var path: Path? = null

    override fun reset() {
        unit = -1
        skill = -1
        target = -1
        path = null
    }
}
